jobrunner/git.py

```diff
@@ -174,12 +177,54 @@ def ensure_git_init(repo_dir):
 
 
 def commit_already_fetched(repo_dir, commit_sha):
+    """
+    Return whether a given commit exists in a repo directory
+
+    We used to do this with:
+
+        git cat-file -e 'COMMIT_SHA^{commit}'
+
+    However it's possible that an interrupted fetch leaves the commit object in
+    place without all of its associated blobs, meaning that the above check
+    passes but attempting to checkout the commit will fail. To work around this
```

Comment:
If we're being pedantic, "checkout" is a noun, and you want "check out".

(cf setup/set up, login/log in, etc... I'm not sure about fuckup/fuck up.)